inlang/source-code/sdk/src/loadProject.ts

```diff
@@ -165,28 +219,56 @@ export async function loadProject(args: {
 				return
 			}
 
-			const loadAndSetMessages = async (fs: NodeishFilesystemSubset) => {
-				makeTrulyAsync(
-					_resolvedModules.resolvedPluginApi.loadMessages({
-						settings: settingsValue,
-						nodeishFs: fs,
-					})
-				)
-					.then((messages) => {
-						setMessages(messages)
-						markInitAsComplete()
-					})
-					.catch((err) => markInitAsFailed(new PluginLoadMessagesError({ cause: err })))
-			}
+			const _settings = settings()
+			if (!_settings) return
 
+			// get plugin finding the plugin that provides loadMessages function
+			const loadMessagePlugin = _resolvedModules.plugins.find(
+				(plugin) => plugin.loadMessages !== undefined
+			)
+
+			// TODO #1844 this watcher needs to get pruned when we have a change in the configs which will trigger this again
 			const fsWithWatcher = createNodeishFsWithWatcher({
 				nodeishFs: nodeishFs,
+				// this message is called whenever a file changes that was read earlier by this filesystem
+				// - the plugin loads messages -> reads the file messages.json -> start watching on messages.json -> updateMessages
 				updateMessages: () => {
-					loadAndSetMessages(nodeishFs)
+					// preserving console.logs as comments pending #
+					// eslint-disable-next-line no-console
+					debug("load messages because of a change in the message.json files")
+					// TODO FINK check error handling for plugin load methods (triggered by file change) -> move to separate ticket
```

Comment:
Create a ticket together with @NiklasBuchfink for this TODO and link it via the issue number.

Reply:
removed TODO (per @NiklasBuchfink all errors are handled and shown in Fink)
added #2358 to improve Fink validation of error handling